Update 0.6.0 release notes with more complete highlights and more PRs (#665)
napari/docs: docs/release/release_0_6_0.md

# napari 0.6.0
⚠️ *Note: these release notes are still in draft while 0.6.0 is in release candidate testing.* ⚠️

Friday, Apr 11, 2025

> *Thu, Apr 10, 2025*

Reviewer: I'm not sure why there's the two dates here?

Author: Oh! Because it's autogenerated from two different files and only one of them is supposed to have the date but I forgot this 😅

## Highlights

### Summary

- Updated viewer handedness ✋

> - Command pallette 🎨

Reviewer 2: ```suggestion
- Command palette 🎨
```

napari! If you run into any issues please get in touch [on GitHub
issues](https://github.com/napari/napari) or on our [Zulip chat room](https://napari.zulipchat.com)!

On the user space, we now offer several options to orient the axes any way you
like:

1. **Through the camera API:** the `Viewer.camera` instance gains two new
  attributes: `orientation`, and `orientation2d`, which is just the last two
  dimensions of `orientation`. You can set the direction that the *depth*,
  *vertical*, and *horizontal* axes point to, respectively in that order, as
  follows ([#7663](https://github.com/napari/napari/pull/7663)):

  ```python
  # 2D
  viewer.camera.orientation2d = ('up', 'right')
  # 3D
  viewer.camera.orientation = ('away', 'up', 'right')
  ```

  See an example of this in action in
  {ref}`sphx_glr_gallery_xarray-latlon-timeseries.py`.

> 2. By right clicking on the dimension toggle in the viewer, and setting the

Reviewer 2: ```suggestion
2. **Through the UI:** By right clicking on the dimension toggle in the viewer, and setting the
```

  axis orientations using the drop-down menus
  ([#7686](https://github.com/napari/napari/pull/7686)), which in 3D will
  further indicate whether the resulting coordinate frame is [right-handed or
  left-handed](https://en.wikipedia.org/wiki/Right-hand_rule)
  ([#7770](https://github.com/napari/napari/pull/7770)):

  ![axis orientation dialog](https://github.com/user-attachments/assets/f73898ec-9156-4f73-ab7f-ee2a7cc17fe1)

> 3. If you want to use a specific axis orientation consistently, you can set the

Reviewer 3: Not a blocker of this alpha, but for the final release we should make clear what the settings were before 0.6.0 vs. the new default, so that a user who wants to go back to the old behavior can do so relatively easily.
See: https://github.com/napari/napari/pull/7787#issuecomment-2794549343

> 3. If you want to use a specific axis orientation consistently, you can set the

Reviewer 2: ```suggestion
3. **Through default startup settings:** If you want to use a specific axis orientation consistently, you can set the
```

  default orientation on startup by changing the relevant settings
  ([#7787](https://github.com/napari/napari/pull/7787):

  ![napari settings panel with axis orientation options highlighted](https://github.com/user-attachments/assets/f5032320-8b03-4ff7-9cb7-8b182ab232af)

### Command palette 🎨

Tired of mousing around? Thanks to
[#5483](https://github.com/napari/napari/pull/5483), napari gains a command
palette! Press {kbd}`Ctrl/Command+Shift+P` and start typing the name of the
action you want to use, and press {kbd}`Enter` when you've highlighted it. It
even works with plugins! This is the culmination of many months of work porting
napari's actions to Talley Lambert's
[app-model](https://github.com/pyapp-kit/app-model). 🥳

(video of command palette)

There's still lots of work to be done here, but in the meantime, give it a try!
We on the team have found it very hard to go back to using napari without the
palette!

### Feature improvements to Shapes layers

⚠️ *In 0.6.0a1 and earlier, this only works when installing
PartSegCore-compiled-backend and toggling the "use compiled triangulation"

> option in the advanced preferences. In 0.6.0 it will work wiht all

Reviewer 4: ```suggestion
option in the advanced preferences. In 0.6.0 it will work with all
```

[#6654](https://github.com/napari/napari/pull/6654)]) Implementing this feature
also eliminated a lot of bugs in our polygon drawing code, which could cause
crashes. If you've had issues with Shapes layers before, now might be a good
time to give them another try!

### Transition to npe2 plugin engine

npe2 was introduced over four years ago, with napari 0.4.12. npe2 has paved the
way for new plugin functionality, such as [adding menu
items](nap-6-contributable-menus) and the command palette. We are now beginning
the process of deprecating npe1 (napari-plugin-engine) plugins, which we need
to do to continue to improve npe2 functionality, for example in file readers,
which is currently very entangled with npe1 code.

To aid this migration, npe1 plugins will now be automatically converted to npe2
by default. This may break some features if the plugins relied on import-time
behavior. ([#7627](https://github.com/napari/napari/pull/7627))

During the 0.6.x series, if some plugin functionality is broken by the
automatic conversion, you can turn off this conversion in the plugin
preferences. However, this option will go away in 0.7.0, which we anticipate to

> happen sometime in July. Therefore, if you encounter conversion issues in a

Reviewer 4: This date genuinely seems too optimistic. Is the goal for plugins to update quickly? Make us panic a bit? Maybe I just don't know the 0.7 roadmap enough

> happen sometime in July. Therefore, if you encounter conversion issues in a

Reviewer 2: ```suggestion
preferences. However, this plugin conversion preferences option will be removed in 0.7.0, which we anticipate to
happen sometime in the second half of 2025. If you encounter conversion issues in a
```

  # 3D
  viewer.camera.orientation = ('away', 'up', 'right')
  ```

  See an example of this in action in
  {ref}`sphx_glr_gallery_xarray-latlon-timeseries.py`.

2. **Through the UI:** By right clicking on the dimension toggle in the viewer,
  and setting the axis orientations using the drop-down menus
  ([#7686](https://github.com/napari/napari/pull/7686)), which in 3D will
  further indicate whether the resulting coordinate frame is [right-handed or
  left-handed](https://en.wikipedia.org/wiki/Right-hand_rule)
  ([#7770](https://github.com/napari/napari/pull/7770)):

  ![axis orientation dialog](https://github.com/user-attachments/assets/f73898ec-9156-4f73-ab7f-ee2a7cc17fe1)

3. **Through the startup settings:** If you want to use a specific axis
  orientation consistently, you can set the default orientation on startup by
  changing the relevant settings
  ([#7787](https://github.com/napari/napari/pull/7787):

  ![napari settings panel with axis orientation options highlighted](https://github.com/user-attachments/assets/f5032320-8b03-4ff7-9cb7-8b182ab232af)

  To restore the orientation from napari 0.5.6 and earlier, change the Depth

>   axis setting to "away" (ie depth axis points away from you).

Reviewer: ```suggestion
  axis setting to "away" (i.e. depth axis points away from you).
```

([#7566](https://github.com/napari/napari/pull/7566),
[#6654](https://github.com/napari/napari/pull/6654)]) Implementing this feature
also eliminated a lot of bugs in our polygon drawing code, which could cause
crashes. If you've had issues with Shapes layers before, now might be a good
time to give them another try!

### Transition to npe2 plugin engine

npe2 was introduced over four years ago, with napari 0.4.12. npe2 has paved the
way for new plugin functionality, such as [adding menu
items](nap-6-contributable-menus) and the command palette. We are now beginning
the process of deprecating npe1 (napari-plugin-engine) plugins, which we need
to do to continue to improve npe2 functionality, for example in file readers,
which is currently very entangled with npe1 code.

To aid this migration, npe1 plugins will now be automatically converted to npe2
by default. This may break some features if the plugins relied on import-time
behavior. ([#7627](https://github.com/napari/napari/pull/7627))

During the 0.6.x series, if some plugin functionality is broken by the
automatic conversion, you can turn off this conversion in the plugin

> preferences. However, the option to turn not convert npe1 plugins will be

Reviewer: ```suggestion
preferences. However, the option to turn off conversion of npe1 plugins will be
```

removed in 0.7.0, and npe1 plugins will *only* work through automatic
conversion. We anticipate 0.7.0 will be released in the second half of 2025.

If you encounter conversion issues in a plugin you rely on, please contact the
plugin authors to encourage them to migrate their plugin to the npe2 system.

> If you are a plugin author and your plugin is not yet npe2-compatible, please

Reviewer: I would add a link to the more [detailed guide](https://napari.org/dev/plugins/advanced_topics/adapted_plugin_guide.html#adapted-plugin-guide) here e.g. 

```suggestion
For more detail on this change and how it affects plugins, see the [detailed guide](adapted-plugin-guide). If you are a plugin author and your plugin is not yet npe2-compatible, please
```

0.6.0. Here are some of the improvements:

- Buttons now have an indicator to show whether they contain an extra menu when
  right-clicking. ([#7556](https://github.com/napari/napari/pull/7556))
- The button to change between 2D and 3D views much more clearly shows
  what it does. ([#7608](https://github.com/napari/napari/pull/7608))
- … And it has an extra menu with lots of options to control the camera!
  ([#7626](https://github.com/napari/napari/pull/7626))
- You can now add a bit of spacing between layers in grid mode (and control it
  in the grid mode right-click menu!)
  ([#7597](https://github.com/napari/napari/pull/7597))
- The colormap indicator in image layers is now a button, allowing you to
  create a linear colormap with any color!
  ([#7600](https://github.com/napari/napari/pull/7600))
- If you select multiple layers in the layer list, you can now see the status
  display of all the selected layers in the status bar
  ([#7673](https://github.com/napari/napari/pull/7673))

### Other stuff

For developers: napari now depends on Python 3.10+ and Pydantic v2.2.

We've supported both pydantic 1 and 2 since 0.4.19, but we're now ready to take

> advantage of performance and API improvements of Pydantic 2. If your library

Reviewer: ```suggestion
advantage of the performance and API improvements of Pydantic 2. If your library
```

- Update conf.py to use packaging to detect dev vs release versions ([docs#631](https://github.com/napari/docs/pull/631))
- Tidy up conf.py file for contributors ([docs#632](https://github.com/napari/docs/pull/632))
- Update troubleshooting.md to address pydantic 2 change in 0.6.0 ([docs#633](https://github.com/napari/docs/pull/633))
- update left sidebar to persist resources ([docs#636](https://github.com/napari/docs/pull/636))
- remove gui_qt since it is deprecated and will be removed in 0.6.0 ([docs#639](https://github.com/napari/docs/pull/639))
- Update napari-workshops.md to include LIBRE LatAm, SciPy, and I2K ([docs#641](https://github.com/napari/docs/pull/641))
- Update event_loop.md to remove gui_qt which is removed in 0.6.0 ([docs#642](https://github.com/napari/docs/pull/642))
- Withdraw NAP-5 new logo proposal ([docs#644](https://github.com/napari/docs/pull/644))
- Update roadmap index to highlight global roadmap  ([docs#651](https://github.com/napari/docs/pull/651))
- Add missing magicgui imports ([docs#655](https://github.com/napari/docs/pull/655))
- Surface plugin best practices ([docs#656](https://github.com/napari/docs/pull/656))

> - Add Tim Monko to core developers list ([docs#662](https://github.com/napari/docs/pull/662))

Reviewer 2: 🎉 🙌🏼 💐

- Update build_and_deploy.yml to use napari docs dependency group ([docs#591](https://github.com/napari/docs/pull/591))
- Replace two manual screenshots of the viewer (launch_cli_empty and launch_cli_image) with nbscreenshots ([docs#606](https://github.com/napari/docs/pull/606))
- Update conf.py to address pygment -> pygments deprecation ([docs#607](https://github.com/napari/docs/pull/607))
- Update CircleCI config to python orb and image ([docs#611](https://github.com/napari/docs/pull/611))
- Use env var to control if notebooks are executed and add docs, slim, slimfast, and slimgallery variants to Makefile ([docs#646](https://github.com/napari/docs/pull/646))
- [bugfix] Fix linkcheck.yml for change to dependencies and removal of req.txt ([docs#654](https://github.com/napari/docs/pull/654))
- Squash deployed gh-pages, relying on this repo for history ([docs#658](https://github.com/napari/docs/pull/658))
- Revert "Squash deployed gh-pages, relying on this repo for history (#658)" ([docs#663](https://github.com/napari/docs/pull/663))

## 19 authors added to this release (alphabetical)

(+) denotes first-time contributors 🥳

> - [andrew sweet](https://github.com/napari/napari/commits?author=andy-sweet) - @andy-sweet

Reviewer: Why did Andy and Clement lose their capital letters :sweat_smile:

Author: 🤔 something weird about their GH profiles. As far as I remember, this is obtained from the GH API... But their profile pages are correct... 🤔 Anyway, will manually fix for now.
